src/Product/VersionChecker.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Text.Json.Serialization;
+
+namespace Azure.DataApiBuilder.Product;
+
+public static class VersionChecker
+{
+    private const string NuGetApiUrl = "https://api.nuget.org/v3-flatcontainer/Microsoft.DataApiBuilder/index.json";
+
+    public static void GetVersions(out string? latestVersion, out string? currentVersion)
+    {
+        latestVersion = FetchLatestNuGetVersion();
+        currentVersion = GetCurrentVersionFromAssembly(Assembly.GetExecutingAssembly());
+    }
+
+    private static string? FetchLatestNuGetVersion()
+    {
+        try
+        {
+            using HttpClient httpClient = new HttpClient { Timeout = TimeSpan.FromSeconds(2) };
+            NuGetVersionResponse? versionData = httpClient.GetFromJsonAsync<NuGetVersionResponse>(NuGetApiUrl)
+                .GetAwaiter().GetResult();
+
+            return versionData?.Versions
+                ?.Where(version => !version.Contains("-rc"))

```

Comment:
Does rc mean that it is not a stable release?

Reply:
Yes, that is correct. RC stands for Release Candidate and those "versions" aren't stable, so we don't want to recommend them to users for upgrading.